alibi/explainers/tests/test_simiarlity/test_grad_methods_integration.py

```diff
@@ -0,0 +1,215 @@
+        current = compute_angle(ds[0], ds[ind])
+        assert current >= last
+        last = current
+
+
+def test_grad_cos_result_order_torch(seed):
+    """
+    `grad_cos` finds data points with small angular separation to be more similar independent of length. Test is
+    applied to `torch` backend.
+    """
+    ds = np.array([[1, 0], [0.9, 0.1], [0.5 * 100, 0.5 * 100]]).astype('float32')
+    model = nn.Linear(2, 1, bias=False)
+    explainer = GradientSimilarity(
+        model,
+        task='regression',
+        loss_fn=loss_torch,
+        sim_fn='grad_cos',
+        backend='pytorch',
+    )
+    explainer = explainer.fit(ds, ds)
+    explanation = explainer.explain(ds[0], Y=ds[0])
+    assert (ds[explanation['ordered_indices'][1]] == ds[1]).all()
+    assert (ds[explanation['ordered_indices'][-1]] == ds[-1]).all()
+
+
+def test_grad_dot_result_order_torch(seed):
+    """
+    Size of datapoint overrides angular closeness for `grad_dot` similarity. Test is applied to `torch` backend.
+    """
+    ds = np.array([[1, 0], [0.9, 0.1], [0.5 * 100, 0.5 * 100]]).astype('float32')
+    model = nn.Linear(2, 1, bias=False)
+    explainer = GradientSimilarity(
+        model,
+        task='regression',
+        loss_fn=loss_torch,
+        sim_fn='grad_dot',
+        backend='pytorch'
+    )
+    explainer = explainer.fit(ds, ds)
+    explanation = explainer.explain(ds[0], Y=ds[0])
+    assert (ds[explanation['ordered_indices'][0]] == ds[-1]).all()
+    assert (ds[explanation['ordered_indices'][-1]] == ds[1]).all()
+
+
+def loss_tf(y, x):
+    return x
+
+
+def test_correct_grad_dot_sim_result_tf(seed, normed_ds):
+    """
+    `grad_dot` method orders data points distributed on the unit circle by their angular separation. Test is applied to
+    `tensorflow` backend.
+    """
+    model = keras.Sequential([keras.layers.Dense(1, use_bias=False)])
+    explainer = GradientSimilarity(
+        model,
+        task='regression',
+        loss_fn=loss_tf,
+        sim_fn='grad_dot',
+        backend='tensorflow'
+    )
+    explainer = explainer.fit(normed_ds, normed_ds)
+    explanation = explainer.explain(normed_ds[0][None], Y=normed_ds[0][None])
+    last = np.dot(normed_ds[0], normed_ds[0])
+    for ind in explanation['ordered_indices'][1:]:
+        current = np.dot(normed_ds[0], normed_ds[ind])
+        assert current <= last
+        last = current
+
+
+def test_correct_grad_cos_sim_result_tf(seed, ds):
+    """
+    `grad_cos` method orders normally distributed data points by their angular separation. Test is applied to
+    `tensorflow` backend.
+    """
+    model = keras.Sequential([keras.layers.Dense(1, use_bias=False)])
+    explainer = GradientSimilarity(
+        model,
+        task='regression',
+        loss_fn=loss_tf,
+        sim_fn='grad_cos',
+        backend='tensorflow'
+    )
+    explainer = explainer.fit(ds, ds)
+    explanation = explainer.explain(ds[0][None], Y=ds[0][None])
```

Comment:
Same comment about extra dimension. If it is not required, just make it consistent